geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -172,29 +275,112 @@ def before_create(table, bind, **kw):
                     col.type, "spatial_index", False
                 ):
                     table.info["_after_create_indexes"].append(idx)
-    _setup_dummy_type(table, gis_cols)
+
+    if db_is_gpkg:
+        if len(gis_cols) > 1:
+            raise ValueError(
+                "Only one geometry column is allowed for a table stored in a GeoPackage."
+            )
+        elif len(gis_cols) == 1:
+            col = gis_cols[0]
+            srid = col.type.srid
+
+            if col.type.geometry_type is None:
+                col.type.geometry_type = "GEOMETRY"
+
+            # Add the SRID of the table in 'gpkg_spatial_ref_sys' if this table exists
+            if (
+                bind.execute(text("""PRAGMA main.table_info("gpkg_spatial_ref_sys");""")).fetchall()
+                and not bind.execute(
+                    text(
+                        """SELECT COUNT(*) FROM gpkg_spatial_ref_sys WHERE srs_id = {};""".format(
+                            srid
+                        )
+                    )
+                ).scalar()
+            ):
+                bind.execute(text("SELECT gpkgInsertEpsgSRID({})".format(srid)))
+            # table.columns = table.info.pop("_saved_columns")
+        _setup_dummy_type(table, gis_cols, gpkg=True)
+    else:
+        _setup_dummy_type(table, gis_cols, gpkg=False)
 
 
 def after_create(table, bind, **kw):
     """Handle spatial indexes during the after_create event."""
     dialect = bind.dialect
     dialect_name = dialect.name
-
-    table.columns = table.info.pop("_saved_columns")
+    db_is_gpkg = is_gpkg(bind)
+
+    if db_is_gpkg:
+        for col in table.columns:
+            # Add the managed Geometry columns with gpkgAddGeometryColumn()
+            if _check_spatial_type(col.type, Geometry, dialect) and check_management(
+                col, dialect_name
+            ):
+                col.type = col._actual_type
+                del col._actual_type
+                dimension = get_col_dim(col)
+                has_z = "Z" in dimension
+                has_m = "M" in dimension
+
+                bind.execute(
+                    text(
+                        """INSERT INTO gpkg_contents
+                        VALUES (
```

Comment:
I think it is always good to explicitly state the column names when inserting. That way you are forwards compatible with possibly added columns.

Reply:
Well, then we are no more forward compatible with renamed columns, which actually already happened in SpatiaLite tables (between v3 and v4 if I remember correctly). So I don't really know what's the best option...